### PR TITLE
Adds social share buttons to the web app #1010

### PR DESCRIPTION
**Pull request by geekyd**
Adds social sharing feature to the generated web app.

@mariobehling @aayusharora @Princu7 Please review.

### CODE REVIEW

File: src/backend/_scss/application.scss [hunk 1]
```diff
@@ -659,7 +659,79 @@ a {
...
+  .title-legend {
+    float: left;
+    width: 40%;
+  }
+
+  .track-names {
+    margin-top: 0;
+  }
+
+  .social-buttons {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: @geekyd Not getting this. According to markup  `social-button class` should be nested  inside `social-buttons`

File: src/backend/_scss/application.scss [hunk 2]
```diff
@@ -927,6 +999,25 @@ a {
...
+    margin-bottom: 0;
+  }
+
+  h3 {
+    display: block;
+  }
+}
+
+@media (max-width: 990px) {
+  .social-container h3 {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: nest `h3` inside.

File: src/backend/_scss/application.scss [hunk 3]
```diff
@@ -659,7 +659,79 @@ a {
...
+      line-height: 30px;
+    }
+  }
+
+  .fa-2x {
+    transform: scale(0.65);
+  }
 }
 
 @media (max-width: 990px) {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: merge this media query with line `1015`

File: src/backend/assets/js/social.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  switch (source) {
+    case 'fb':
+      window.open('http://www.facebook.com/share.php?u=' + window.location, 'facebook', 'height=360, width=580');
+      break;
+    case 'tw':
+      window.open('http://twitter.com/home?status=' + window.location, 'twitter', 'height=360, width=580');
+      break;
+    case 'go':
+      window.open('https://plus.google.com/share?url=' + encodeURIComponent(window.location), 'google', 'height=360, width=580');
```
Comment (aayusharora) [inline src/backend/assets/js/social.js]: why do we need encodeURIcomponent here.

Comment (geekyd) [inline src/backend/assets/js/social.js]: Google plus was not working without encoding the url. So I added the it to encode the url.

File: src/backend/_scss/application.scss [hunk 5]
```diff
@@ -659,10 +659,47 @@ a {
 
 .track-names {
   color: $black;
-  margin-top: 16%;
+  margin-top: 0;
+}
+
+.social-buttons {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: @geekyd Why there is another `.social-buttons` class defined on line 712. Please remove this one.

File: src/backend/assets/js/social.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+$('.social-button').click(function() {
+  var source = this.id.toString().split('-')[0];
+
+  if (source === 'fb') {
+    window.open('http://www.facebook.com/share.php?u=' + window.location, 'facebook', 'height=360, width=580');
+  } else if (source === 'tw') {
+    window.open('http://twitter.com/home?status=' + window.location, 'twitter', 'height=360, width=580');
+  } else if (source === 'go') {
+    window.open('https://plus.google.com/share?url=' + encodeURIComponent(window.location), 'google', 'height=360, width=580');
```
Comment (aayusharora) [inline src/backend/assets/js/social.js]: What is the use of `encodeURIComponent` here.

File: src/backend/_scss/application.scss (hunk 5 above)
Comment (geekyd) [inline src/backend/_scss/application.scss]: .social-buttons on 712 is inside `@media`

File: src/backend/assets/js/social.js (hunk 6 above)
Comment (geekyd) [inline src/backend/assets/js/social.js]: After testing the build on heruku, the url without `encodeURIComponent ` was invalid url for gplus, after adding `encodeURIComponent ` the url was no more invalid.

File: src/backend/_scss/application.scss (hunk 5 above)
Comment (aayusharora) [inline src/backend/_scss/application.scss]: @geekyd, In that case make a mixin that can be @include in both of them.

File: src/backend/assets/js/social.js (hunk 6 above)
Comment (aayusharora) [inline src/backend/assets/js/social.js]: @geekyd Check this too.

Comment (aayusharora) [inline src/backend/assets/js/social.js]: If it is due to special characters, it should be same for all fb and twitter links as well.

Comment (geekyd) [inline src/backend/assets/js/social.js]: @aayusharora I have noticed the url of the share window is auto encoded only for twitter, and not for facebook and google plus. So special characters might be buggy for unencoded urls.

File: src/backend/_scss/application.scss (hunk 5 above)
Comment (geekyd) [inline src/backend/_scss/application.scss]: @aayusharora mixin is good idea but the tag at `665` and `712` are not repeated code.

File: src/backend/assets/js/social.js (hunk 6 above)
Comment (geekyd) [inline src/backend/assets/js/social.js]: **Without Encoding**
screen shot 2017-02-20 at 3 05 00 pm


**With Encoding**

screen shot 2017-02-20 at 3 05 41 pm

Comment (aayusharora) [inline src/backend/assets/js/social.js]: ok Agree !!

**Comment by mariobehling** [comment]
Good idea, but this implementation is not in line with the design of the site. I need to revert it therefore. Please propose a design implementation that is according to the existing design.

**Pull request merged**